client/src/main/java/io/pravega/client/netty/impl/ClientConnectionImpl.java

```diff
@@ -49,20 +53,43 @@ public ClientConnectionImpl(String connectionName, int flowId, FlowHandler netty
     public void send(WireCommand cmd) throws ConnectionFailedException {
         checkClientConnectionClosed();
         nettyHandler.setRecentMessage();
-        Futures.getAndHandleExceptions(nettyHandler.getChannel().writeAndFlush(cmd), ConnectionFailedException::new);
+        write(cmd);
     }
 
     @Override
     public void send(Append append) throws ConnectionFailedException {
         Timer timer = new Timer();
         checkClientConnectionClosed();
         nettyHandler.setRecentMessage();
-        Futures.getAndHandleExceptions(nettyHandler.getChannel().writeAndFlush(append), ConnectionFailedException::new);
+        write(append);
         nettyHandler.getMetricNotifier()
                     .updateSuccessMetric(CLIENT_APPEND_LATENCY, segmentTags(append.getSegment(), append.getWriterId().toString()),
                                          timer.getElapsedMillis());
     }
 
+    private void write(Object cmd) throws ConnectionFailedException {
+        Channel channel = nettyHandler.getChannel();
+        EventLoop eventLoop = channel.eventLoop();
+        ChannelPromise promise = channel.newPromise();
+        promise.addListener(ChannelFutureListener.FIRE_EXCEPTION_ON_FAILURE);
+        AtomicReference<ChannelFuture> future = new AtomicReference<>();
+        if (eventLoop.inEventLoop()) {
+            channel.write(cmd, promise);
```

Comment:
we can use a void promise.

Reply:
I need to set the listener on the promise to make sure the error handler is invoked.